app/src/main/java/com/github/studydistractor/sdp/data/Chat.kt

```diff
@@ -0,0 +1,25 @@
+package com.github.studydistractor.sdp.data
+
+/**
+ * Represent a chat (place where users can exchange messages)
+ *
+ * @param chatId id of the chat
+ * @param messages list of messages that were sent in the chat
+ */
+data class Chat(
+    val chatId: String? = null,
+    val messages: List<Message> = listOf()
+)
+
+/**
+ * Represent a message that is sent in the chat
+ *
+ * @property messageId id of the message
+ * @property timeStamp timestamp of the message (time when the message was sent)
+ * @property userId id of the user that sent the message
+ */
+data class Message(
+    val messageId: String? = null,
+    val timeStamp: String? = null,

```

Comment:
why is the timestamp optionnaaaaal

Reply:
From [Firebase docs](https://firebase.google.com/docs/database/android/read-and-write) : 
*Null default values create a no-argument default constructor, which is needed for deserialization from a DataSnapshot.*

if someone can check if 'val messageId: String = ""' can be accepted for deserialization. Otherwise, we need to keep the null values. (I ran into silly bugs before because of this before)

Reply:
Ugh, this is ugly.

Maybe we could have two data classes : 
```kotlin
data class Message(
    val messageId: String,
    val timeStamp: String,
    ...
)

data class FirebaseMessage(
    val messageId: String?,
    val timeStamp: String?,
    ...

    fun toMessage(): Message {
        // convert to a Message with non-nullable attributes (with a simple nullity check of each parameter
    }
)
```

But idk if that's too complicated. I think we should do it but like get other people's opinion.

Reply:
At the end of the day, we should check the data when we receive them from the service. And I prefer having to deals with nulls rather than parsing and checking if the string is valid

Reply:
I also think it's somewhat nice to separate firebase data structures from actual data structures, that way the data structure can be easily ported to a different db schema or a different db altogether

Reply:
I agree with @aizensoosuke

Reply:
I also think it's better to have the two data classes like @aizensoosuke  suggests, it doesn't seem to complicated.